### PR TITLE
squotable

### PR DESCRIPTION
**Pull request by ethdmill**
- added squotable command
- updated other files as necessary

### CODE REVIEW

File: commands/hello.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const HelloCommand: Command = {
       message.channel.send(`Oh ... hey <@${message.author.id}> ${emoji}`);
     } else {
       const emoji = randomElement(HappyEmoji);
-      message.channel.send(`${capitalize(command[0])}, <@${message.author.id}>! ${emoji}`);
+      message.channel.send(`${capitalize(command)}, <@${message.author.id}>! ${emoji}`);
```
Comment (mattbalmer) [inline commands/hello.ts:42]: Thanks for this!

Comment (mattbalmer) [inline commands/hello.ts:42]: Thanks for this!

File: commands/most-squotable.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import * as Discord from 'discord.js';
+import { Message, GuildChannel, TextChannel } from 'discord.js';
+import { Command } from 'squidly/types';
+import { mostFrequentElement } from 'squidly/utils/arrays';
+import { EMBED_COLORS } from 'squidly/constants/colors';
+
+const Keywords = [
+  'squotable',
+  'squoteable',
+  'most_squotable',
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Can probably delete these underscore commands, tbh

File: commands/most-squotable.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  'most_squoteist',
+  'top_squoter',
+  'best_squoter',
+  'funniest_member',
+  'funniest_user'
+];
+
+export const MostSquotable: Command = {
+  shouldHandle: (message, { command }) =>
+  Keywords.includes(command),
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Indent here, see Hello.ts

Will add a linter at some point, sorry

File: commands/most-squotable.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  
+  handle: async (message: Message, { tag, command, args }) => {
+    const squotesChannel = message.guild.channels.cache.find((channel: GuildChannel) => {
+      if (channel.name === "squotes") {
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Let's add "squotes" as a configurable constant to ENV, then reference it here. (similar to GAME_CHANNEL_NAME etc)

File: commands/most-squotable.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  'funniest_member',
+  'funniest_user'
+];
+
+export const MostSquotable: Command = {
+  shouldHandle: (message, { command }) =>
+  Keywords.includes(command),
+  
+  handle: async (message: Message, { tag, command, args }) => {
+    const squotesChannel = message.guild.channels.cache.find((channel: GuildChannel) => {
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Can reduce this to one line
```suggestion
const squotesChannel = message.guild.channels.cache.find(channel => channel.name === ENV.QUOTES_CHANNEL) as TextChannel;
```

File: commands/most-squotable.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      if (channel.name === "squotes") {
+        return true;
+      };
+    }) as TextChannel;
+
+    const messageCollection = await squotesChannel.messages.fetch();
+    const squotesMessages = messageCollection.array();
+    let mentionedDisplayNames: string[] = [];
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Gonna change this to collect actual GuildMember objects, will discuss offline later how to optimize.

File: commands/most-squotable.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  
+  handle: async (message: Message, { tag, command, args }) => {
+    const squotesChannel = message.guild.channels.cache.find((channel: GuildChannel) => {
+      if (channel.name === "squotes") {
+        return true;
+      };
+    }) as TextChannel;
+
+    const messageCollection = await squotesChannel.messages.fetch();
+    const squotesMessages = messageCollection.array();
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Let's use the Collection's built-in `.forEach`,, no need to `.array()` here

File: commands/most-squotable.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        let lastMentionedMember = currentMembersMentioned[currentMembersMentioned.length - 1];
+        mentionedDisplayNames.push(lastMentionedMember.displayName);
+      };
+    };
+
+    let mostSquotedMember = mostFrequentElement(mentionedDisplayNames);
+
+    message.channel.send(new Discord.MessageEmbed()
+        .setColor(EMBED_COLORS.SUCCESS)
+        .setDescription(`Looks as though people think ${mostSquotedMember} is the funniest member of Spanksquad! Try harder, everyone else!`)
```
Comment (mattbalmer) [inline commands/most-squotable.ts]: Thoughts on adding different formats for the reply, then picking one at random?

File: utils/arrays.ts [hunk 9]
```diff
@@ -2,4 +2,25 @@ import { random } from 'squidly/utils/math';
 
 export function randomElement(array) {
   return array[Math.floor(random() * array.length)];
-}
\ No newline at end of file
+};
+
+export function mostFrequentElement(array) {
```
Comment (mattbalmer) [inline utils/arrays.ts]: Will also want to make this more generic to handle non-primitives, can discuss offline.

Thanks for the addition though!

**Review by mattbalmer — CHANGES_REQUESTED** [review]
Looks good overall! Some specific recommendations for naming & abstraction + one thought on command experience.

File: commands/most-squotable.ts (hunk 2 above)
Comment (ethdmill) [reply commands/most-squotable.ts]: Will delete upon resolving other suggestions so as to not lose too many line number references

File: commands/most-squotable.ts (hunk 3 above)
Comment (ethdmill) [reply commands/most-squotable.ts]: Indented

File: commands/most-squotable.ts (hunk 8 above)
Comment (ethdmill) [reply commands/most-squotable.ts]: Added new responses!

File: commands/most-squotable.ts (hunk 6 above)
Comment (ethdmill) [reply commands/most-squotable.ts]: Marking resolved for now per offline conversation

File: commands/most-squotable.ts (hunk 7 above)
Comment (ethdmill) [reply commands/most-squotable.ts]: Discussed offline

File: utils/arrays.ts (hunk 9 above)
Comment (ethdmill) [reply utils/arrays.ts]: Discussed offline

**Review by mattbalmer — APPROVED** [review]
Looks good, thanks for this!

**Pull request merged**